lib/LedgerEventStorage.js

```diff
@@ -159,10 +159,16 @@ class LedgerEventStorage {
           if(!database.isDuplicateError(e)) {
             throw e;
           }
-          // FIXME check that error still has index
-          // remove events up to the dup and retry
-          dupHashes.push(chunk[e.index].meta.eventHash);
-          chunk = chunk.slice(e.index + 1);
+          // if for some reason these are no writeErrors throw
+          if(!e.writeErrors) {
+            throw e;
+          }
+          // if there are writeErrors then correct for them
+          for(const bulkError of e.writeErrors) {
```

Comment:
When `ordered` is true, I don't think this is necessary. Please run a simple test that inserts several records where one of them in the middle is a duplicate and see what's in the response. We may be able to avoid the complexity here.

Reply:
Alternatively, if we discover that `ordered: false` can be used here, we can do that instead, because I believe it would insert all non-duplicates with a single call. Then we could just check the errors to make sure all of them were duplicate errors.

@mattcollier -- Do you remember if `ordered: true` was needed? Are the events in a particular order such that if not inserted in that order we could have corruption or some irrecoverable error?

Reply:
Yes, `ordered` is **CRITICAL**.  If we lose this guarantee consistency on the ledger is compromise.

Reply:
I do believe that using `ordered:false` is possible now: https://docs.mongodb.com/manual/reference/method/db.collection.bulkWrite/

If you want me to, I can switch this to false and try it.
I'll build the 5 docs with one in the middle being a duplicate in a moment.

Reply:
@aljones15, please build the example with `ordered: true` -- @mattcollier has confirmed that ordered MUST be true here.

Reply:
ok added the test and it is working as expected:

https://github.com/digitalbazaar/bedrock-ledger-storage-mongodb/pull/53/commits/89d764625bb05fc25c0268c35d0953a7d6505c15

Reply:
This is what the error looks like: https://github.com/digitalbazaar/bedrock-ledger-storage-mongodb/issues/52#issuecomment-705619648